sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -278,13 +278,41 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
   }
 
   def ansiDateTimeError(e: Exception): SparkDateTimeException = {
+    def extractDateTimeErrorInfo(e: Exception): (String, String, String) = {
+      val errorMessage = e.getMessage
+
+      val valuePattern = "Invalid value for ([A-Za-z]+) \\(valid values (.+)\\): (.+)".r
+      val datePattern = "Invalid date '[A-Z]+ ([0-9]+)'".r
+
+      errorMessage match {
+        case valuePattern(field, range, badValue) =>
+          val unit = field match {
+            case "Year" => "YEAR"
+            case "MonthOfYear" => "MONTH"
+            case "DayOfMonth" => "DAY"
+            case "HourOfDay" => "HOUR"
+            case "MinuteOfHour" => "MINUTE"
+            case "SecondOfMinute" => "SECOND"
+          }
+          val formattedRange = range.replace(" - ", " ... ")
+          (unit, formattedRange, badValue)
+        case datePattern(badDate) =>
+          ("DAY", "1 ... 28/31", badDate)
+      }
```

Comment:
@itholic Do you expect that it should match to both the cases above. What if not? I would add a default case.

Reply:
For example, `toInstant` can raise:
```
        if (seconds < MIN_SECOND || seconds > MAX_SECOND) {
            throw new DateTimeException("Instant exceeds minimum or maximum instant");
        }
```

How does your code handles this?

Reply:
Good point.

I think `DATETIME_FIELD_OUT_OF_BOUNDS ` is not very proper error class to cover the example case, so let me leave the existing `_LEGACY_ERROR_TEMP_2000` as it is for default case.

We may need introduce several new error classes to cover the all potential exception cases along with more test cases in a separate ticket.